x/ecocredit/basket/date_criteria.go

```diff
@@ -35,16 +35,16 @@ func (d *DateCriteria) Validate() error {
 	if (minStartDate != nil && startDateWindow != nil) ||
 		(startDateWindow != nil && yearsInThePast != 0) ||
 		(minStartDate != nil && yearsInThePast != 0) {
-		return fmt.Errorf("only one of min_start_date, start_date_window, or years_in_the_past must be set")
+		return sdkerrors.ErrInvalidRequest.Wrap("only one of min_start_date, start_date_window, or years_in_the_past must be set")
```

Comment:
The `invalid request` error gets repeated if we add it here. We currently use `ErrInvalidRequest` when validating the message and in general I think we should avoid using sdk errors in helper functions such as this one.

https://github.com/regen-network/regen-ledger/runs/6771149123?check_suite_focus=true#step:5:1288

```
invalid date criteria: only one of min_start_date, start_date_window, or years_in_the_past must be set: invalid request: invalid request
```